internal/services/batch/batch_account_resource.go

```diff
@@ -311,15 +309,18 @@ func resourceBatchAccountUpdate(d *pluginsdk.ResourceData, meta interface{}) err
 
 	parameters := batch.AccountUpdateParameters{
 		AccountUpdateProperties: &batch.AccountUpdateProperties{
-			AutoStorage: &batch.AutoStorageBaseProperties{
-				StorageAccountID: &storageAccountId,
-			},
 			Encryption: encryption,
 		},
 		Identity: identity,
 		Tags:     tags.Expand(t),
 	}
 
+	if v, ok := d.GetOk("storage_account_id"); ok && v.(string) != "" {
```

Comment:
this field is Optional & Computed, meaning there's no way to remove this with this configuration - can we add a test which explicitly removes the storage account id?

Reply:
Thanks for your feedback. After verifying the API, the `storage_account_id` does not need to be computed. So, I removed the computed of `storage_account_id`. For more information, please see code updates.